pkgs/applications/networking/cluster/k3s/default.nix

```diff
@@ -1,16 +1,23 @@
-{ lib, stdenv, callPackage }:
+{ lib, callPackage, ... }@args:
 
 let
   k3s_builder = import ./builder.nix lib;
   common = opts: callPackage (k3s_builder opts);
+  # extraArgs is the extra arguments passed in by the caller to propogate downward.
+  # This is to allow all-packages.nix to do:
+  #
+  #     let k3s_1_23 = (callPackage ./path/to/k3s {
+  #       commonK3sArg = ....
+  #     }).k3s_1_23;
+  extraArgs = lib.filterAttrs (n: v: !(lib.elem n [ "callPackage" ]) ) args;
```

Comment:
There might be a better way to do this; I'm open to pointers to other instances of this pattern or ideas.

I do think it'd be nice for the `callPackage` usage at the top-level to look clean, and this was the simplest change I could think of to make the usual pattern there work as expected!